collectors/python.d.plugin/alarms/alarms.chart.py

```diff
@@ -0,0 +1,51 @@
+# -*- coding: utf-8 -*-
+# Description: alarms netdata python.d module
+# Author: andrewm4894
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from json import loads
+
+from bases.FrameworkServices.UrlService import UrlService
+
+priority = 200200
+update_every = 5
+disabled_by_default = True
+
+DEFAULT_STATUS_MAP = {'CLEAR': 0, 'WARNING': 1, 'CRITICAL': 2}
+
+ORDER = [
+    'alarms',
+]
+
+CHARTS = {
+    'alarms': {
+        'options': ['alarms', 'Alarms', 'status', 'alarms', 'alarms.status', 'line'],
+        'lines': []
+    }
+}
+
+
+class Service(UrlService):
+    def __init__(self, configuration=None, name=None):
+        UrlService.__init__(self, configuration=configuration, name=name)
+        self.order = ORDER
+        self.definitions = CHARTS
+        self.status_map = self.configuration.get('status_map', DEFAULT_STATUS_MAP)
+
+    def validate_charts(self, name, data, algorithm='absolute', multiplier=1, divisor=1):
+        for dim in data:
+            if name not in self.charts:
+                chart_params = [name] + CHARTS[name]['options']
+                self.charts.add_chart(params=chart_params)
+            if dim not in self.charts[name]:
+                self.charts[name].add_dimension([dim, dim, algorithm, multiplier, divisor])
+
+    def _get_data(self):
+        raw_data = self._get_raw_data()
+        if raw_data is None:
+            return None
+        raw_data = loads(raw_data)
+        alarms = raw_data.get('alarms', {})
+        data = {a: self.status_map[alarms[a]['status']] for a in alarms if alarms[a]['status'] in self.status_map}
```

Comment:
Does it really work? i assume right after the start of netdata there could be 0 alarms in some states and `_get_data` would return an empty dict (job fails).

Reply:
Yeah - so long as you have `CLEAR` in the `status_map` i think it will be ok. 

E.g on  startup i see something like this (all alarms in CLEAR state):

![image](https://user-images.githubusercontent.com/2178292/99979128-b504b780-2d9e-11eb-9b2f-a30062927d9c.png)

And then if i have some alarms i see:

![image](https://user-images.githubusercontent.com/2178292/99979368-ee3d2780-2d9e-11eb-8db8-f5efcf10879c.png)

So when I get some alarms it seems the 0 valued dimensions disappear.

Reply:
I checked, first response right after the restart is

```cmd
2020-11-23 19:00:07: python.d INFO: alarms[alarms] : {'hostname': 'pc', 'latest_alarm_log_unique_id': 1606139165, 'status': True, 'now': 1606147207, 'alarms': {}}
```

Reply:
Fix is to add `alarms_num` variable to the chart, so we never return an empty `data` on 200 response